pingcap/tidb
# *: refine transaction related log

### What problem does this PR solve? 
Make it easier to address bug.

### What is changed and how it works?
- use prettyPrintKey for Lock.String()
- log the event when LockResolver rollbacks a key.
- remove the warning log when failed to extract `conflictCommitTs`

The new resolveLock rollback log looks like 
```
[2019/11/07 12:57:20.515 +00:00] [INFO] [lock_resolver.go:524] ["resolveLock rollback"] [lock="key: {tableID=51, handle=6}, primary: {tableID=51, handle=0}, txnStartTS: 412386968233312276, ttl: 3281, type: Put"]
```

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
 run jepsen test case and see what the new log looks like.


Related changes

 - Need to cherry-pick to the release branch
